crates/ty/src/main.rs

```diff
@@ -228,22 +228,22 @@ impl MainLoop {
 
         self.watcher = Some(ProjectWatcher::new(watcher, db));
 
-        self.run(db)?;
+        self.run(db, true)?;
 
         Ok(ExitStatus::Success)
     }
 
-    fn run(mut self, db: &mut ProjectDatabase) -> Result<ExitStatus> {
+    fn run(mut self, db: &mut ProjectDatabase, watch: bool) -> Result<ExitStatus> {
```

Comment:
Nit: Should we have a `run` and a `run_with_progress` method. It would reduce the amount of changes necessary (and you could implement `Reporter` for `()`)

Reply:
That's fine with me too